builtin/logical/database/path_config_connection.go

```diff
@@ -0,0 +1,274 @@
+package database
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/vault/builtin/logical/database/dbplugin"
+	"github.com/hashicorp/vault/helper/strutil"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+var (
+	respErrEmptyPluginName = logical.ErrorResponse("empty plugin name")
```

Comment:
This returns a `*logical.Response` which in theory could be modified later in e.g. `core` because it's not expecting anything/anyone else to care. I'd probably be better to make this a string that can be plugged in as the message for a new `logical.ErrorResponse` invocation; or maybe even a discrete error type (or at least discrete error via `errors.New`) whose `.Error()` method can be used. Either way, this could then also get moved to pluginutil which is probably where it belongs?